### PR TITLE
ddl: Allow binary UUID as column default

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Allow binary UUID as column default

```
sql> CREATE TABLE t1 (id INT PRIMARY KEY, u VARBINARY(16) DEFAULT (UUID_TO_BIN(UUID())));
Query OK, 0 rows affected (0.0267 sec)

sql> INSERT INTO t1(id) VALUES(1),(2),(3);
Query OK, 3 rows affected (0.0108 sec)

Records: 3  Duplicates: 0  Warnings: 0

sql> SELECT id, BIN_TO_UUID(u) FROM t1;
+----+--------------------------------------+
| id | BIN_TO_UUID(u)                       |
+----+--------------------------------------+
|  1 | 827a6ab8-9b37-11ee-9127-5405db7aad56 |
|  2 | 827a7e46-9b37-11ee-9127-5405db7aad56 |
|  3 | 827a7f0a-9b37-11ee-9127-5405db7aad56 |
+----+--------------------------------------+
3 rows in set (0.0026 sec)

sql>
```

Issue Number: ref #33870 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
The `UUID_TO_BIN()` function can now be used as default for a column
```

### CODE REVIEW

**Review by tangenta — COMMENTED** [review]
Could you add a test?

**Comment by dveeden** [comment]
> Could you add a test?

Yes. That was also the reason this was set as a draft.

**Comment by CbcWestwolf** [comment]
LGTM

**Review by CbcWestwolf — APPROVED** [review]

File: pkg/ddl/db_integration_test.go [hunk 1]
```diff
@@ -1606,6 +1606,55 @@ func TestDefaultColumnWithRand(t *testing.T) {
 	tk.MustGetErrCode("CREATE TABLE t3 (c int, c1 int default a_function_not_supported_yet());", errno.ErrDefValGeneratedNamedFunctionIsNotAllowed)
 }
 
+func TestDefaultColumnWithUUID(t *testing.T) {
```
Comment (zimulala) [inline pkg/ddl/db_integration_test.go:1609]: Do we need to put this test to `tests/integrationtest/t/ddl/db_integration.test`? @Defined2014

Comment (Defined2014) [reply pkg/ddl/db_integration_test.go:1609]: Better do this, thx 🌹

Comment (dveeden) [reply pkg/ddl/db_integration_test.go:1609]: Tried to do this in db9f97f4cbb15f5e0837e92c2e19ca1591cef88c

@Defined2014 , @zimulala is this what you suggested?

Comment (Defined2014) [reply pkg/ddl/db_integration_test.go:1609]: Yes, and I think you could delete this test now. Integrationtest will run in both unistore and TiKV env.

Comment (dveeden) [reply pkg/ddl/db_integration_test.go]: done

**Review by Defined2014 — APPROVED** [review]
LGTM
/cc @zimulala

**Review by zimulala — APPROVED** [review]

**Pull request merged**